pylint/checkers/strings.py

```diff
@@ -373,11 +374,11 @@ def visit_binop(self, node):
             # Check that the number of arguments passed to the RHS of
             # the % operator matches the number required by the format
             # string.
-            args_elts = ()
+            args_elts = []
             if isinstance(args, nodes.Tuple):
                 rhs_tuple = utils.safe_infer(args)
                 num_args = None
-                if hasattr(rhs_tuple, "elts"):
+                if rhs_tuple and hasattr(rhs_tuple, "elts"):
```

Comment:
We should not use `hasattr`. `isinstance(rhs_tuple, nodes.BaseContainer)` would be better. With that the `if rhs_tuple` check will be redundant too.

Reply:
Done!